projects/hathor-wallet-headless/0001-multisig-create-mint-melt-token.md

```diff
@@ -0,0 +1,327 @@
+
+With this specification we cover the possibility to melt tokens, send it to another wallet, create or not another melt authority, and send the authority to another wallet.
+
+## Inspect transaction with wallet metadata
+
+There is an endpoint to [inspect the transaction](https://hathor.gitbook.io/hathor/guides/multisig-wallet/sending-transactions#5.-check-the-multisig-transaction) by decoding the transaction hexadecimal encode in `POST:/wallet/decode`. However, it doesn't bring signature information, nor wallet information with it.
+
+For the MultiSig wallet we have the `inspect` command:
+`POST:/wallet/p2sh/tx-proposal/inspect`
+
+The data field is:
+* txHex
+
+The response scheme for success:
+```ts
+{
+  "success": boolean,
+  "complete": boolean,
+  "tx": {
+    "version": number,
+    "tokens": string[],
+    "inputs": [
+      {
+        tx_id: string,
+        index: number,
+        decoded: {},  // data from the spent ouput
+        signed: boolean,
+        mine: boolean,
+      },
+    ],
+    "outputs": [
+      {
+        value: number,
+        token_data: number,
+        script: string,
+        decoded: {},
+        authorities: number,
+        mine: boolean,
+      }
+    ],
+  },
+  "balance": {
+    "<token-uid>":  {
+      "tokens": { available: number, locked: number ),
+      "authorities": {
+        mint: { available: number, locked: number ),
+        melt: { available: number, locked: number ),
+      },
+    },
+  },
+}
+```
+
+Lets represent the response document as ` $ `, as each element of a list ` [*] `, we have:
+* `$.complete` -- that represents the completeness of signatures required to use the transaction
+* `$.tx.inputs[*].signed` -- that indicates this input has a signature
+* `$.tx.inputs[*].mine` -- that indicates this input belongs to this wallet
+* `$.tx.outputs[*].mine` -- that indicates this output belongs to this wallet
+* `$.balance.tokens` -- that contains the balance of each token in the transaction
+* `$.balance.authorities` -- that contains the balance of mint and melt for each token
+
+## Initialize MultiSig wallet without seed
+
+This wallet is used by the coordinator service to sign or sign and push the transaction given the signatures collected by the service.
+
+By relaxing the checks in `/wallet/start` endpoint logic a wallet can have in its configuration the `multisig` key without the `seeds` key. This configuration impairs the wallet to use `/wallet/p2sh/tx-proposal/get-my-signatures` endpoint, however it still allows the use of `/wallet/p2sh/tx-proposal/sign` and `/wallet/p2sh/tx-proposal/sign-and-push` endpoints because the signatures are provided in the call.
+
+This wallet can also be called the "seedless wallet".
```

Comment:
I would like to understand a bit the low level of this feature. The sign endpoint calls a method that validates the utxos of the multisig wallet, for instance, so you would need to create a readonly wallet with balance, history and utxos loaded, not only a seedless wallet. That's why I feel a better description of the low level would be good. The lib already supports a readonly wallet for multisig?

Reply:
The idea is that we only require the multisig configuration to derive the addresses, so we can know the balance of the wallet and receive transactions in real time.
If we receive the signatures from the actual participants and we have the multisig configuration we can derive the redeem script and input data, making the transaction ready to be mined.

But I agree the design should include how we are going to do this.

Reply:
Working on it.

Reply:
I found that we don't need change anything. I updated the topic "Initialize read-only MultiSig wallet".